source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -4443,6 +4455,65 @@ export class ShowCardAction extends Action {
     }
 }
 
+export class OverflowAction extends Action {
+    static readonly JsonTypeName: "Action.Overflow" = "Action.Overflow";
+    private contextMenu: PopupMenu;
+    private readonly styleNodeId = "ac-ctrlContextMenu";
+
+    constructor(
+        private actions: Action[]
+	) {
+        super();
+		this.title = Strings.defaults.overflowButtonText();
+    }
+
+    getActions(): readonly Action[] {
+        return this.actions;
+    }
+
+    getJsonTypeName(): string {
+        return ShowCardAction.JsonTypeName;
+    }
+
+    execute() {
+        const shouldDisplayPopupMenu = raiseDisplayOverflowActionMenuEvent(this, this.renderedElement);
+        if (shouldDisplayPopupMenu && this.renderedElement) {
+            if (!this.contextMenu) {
+                this.contextMenu = new PopupMenu();
+                this.contextMenu.onClose = () => {
+                    this.removeCssForContextMenu();
+                }
+                this.actions.forEach((action, id) => {
+                    const menuItem = new DropDownItem(id.toString(), action.title ?? "");
+                    menuItem.onClick = () => {
+                        action.execute();
+                        this.contextMenu.closePopup(false);
+                    };
+                    this.contextMenu.items.add(menuItem);
+                });
+            }
+
+            this.setupCssForContextMenu();
+            this.contextMenu.popup(this.renderedElement);
+        }
+	}
+
+    private setupCssForContextMenu() {
```

Comment:
Thinking more about CSS. We actually need to move all the control styles into the main AC CSS (see one of my last comments in this review for more details.) Thus, it will be th responsibility of the host application to load the appropriate CSS file (with its own customization) and we do not need to dynamically add/remove that `style` node.